### PR TITLE
Keep associative-memory utility finite past inf-inf cross-entropy

### PR DESCRIPTION
**Pull request by Kayvan-Zahiri**
## Summary
- Cross-entropy on inf logits is `inf - inf = NaN`. That NaN is clipped into stored utility, and `argmin` then treats the NaN row as the lowest-utility occupant.
- Fail-closed: non-finite logits report uniform loss. Non-finite utility updates keep the previous finite value. Replacement ignores non-finite utilities.
- Finite rows still compete normally.

## Test plan
- [x] `.venv/bin/python -m pytest tests/test_associative_memory.py -q`
- [x] `.venv/bin/python -m ruff check alberta_framework/core/associative_memory.py tests/test_associative_memory.py`

No Slop measured-run receipt. Made with Cursor.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by Prabhat1308 — APPROVED** [review]
Independent review of head `eeaac17fb7b44d7d0921d4263a39db3602e505ad` against `main` `da8b86c`, under the `review-asi-contributions` skill. Head tree verified as current `main` plus exactly `alberta_framework/core/associative_memory.py` + `tests/test_associative_memory.py`; automated adversarial scan clean (no dependency, CI, hook, or lockfile changes; no assertions or raises removed anywhere in the fleet).

**Reproduced:** `tests/test_associative_memory.py` at head — **11 passed in 3.46s**; repo-wide Ruff: All checks passed!; strict mypy: Success: no issues found in 159 source files; `git diff --check`: clean. Red phase with `alberta_framework/core/associative_memory.py` reverted to `main`, PR tests kept: **2 failed, 9 passed in 3.37s** (2 new regression test(s) fail), confirming failing-test-first.

**Scope note:** the guard skips updates only on non-finite inputs (or non-finite derived diagnostics); finite paths are numerically unchanged. Later finite samples resume learning from the last finite state.

Process context, stated factually: this PR is part of a 27-PR same-day fleet by one author in the same non-finite-guard class, with honest Cursor authorship disclosure, no measured-run receipt, and no linked pre-registration issue; receipts cannot create score and their absence does not gate correctness — flagged only so the maintainer weighs the fleet, not just each PR.

Recommendation: **accept**.

```slop-review
{"schemaVersion": "1", "projectId": "asi", "artifactUrl": "https://github.com/elizaOS/asi/pull/90", "headSha": "eeaac17fb7b44d7d0921d4263a39db3602e505ad", "recommendation": "accept", "reproduced": true, "securityRisk": "none", "duplicateRisk": "none", "usefulArtifacts": ["non-finite guard in alberta_framework/core/associative_memory.py with failing-test-first regression test(s), reproduced red on main (2 failed, 9 passed in 3.37s) and green at head (11 passed in 3.46s)"], "commands": [".venv/bin/python -m pytest tests/test_associative_memory.py -q -o addopts=\"\"", ".venv/bin/python -m ruff check .", ".venv/bin/python -m mypy", "git diff --check"], "evidenceUrls": ["https://github.com/elizaOS/asi/pull/90"], "summary": "Head verified as main plus the stated two-file diff; red/green reproduced (2 regression test(s) red on main, 11 passed in 3.46s at head). Part of a 27-PR same-day non-finite-guard fleet, disclosed Cursor-authored, no receipts, no linked issues."}
```
